### PR TITLE
Add an experiment framework.

### PR DESCRIPTION
**Pull request by shans**
This PR also contains a small improvement to the CSSModel. LMK if you'd rather that was submitted separately.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -86,28 +99,104 @@ gulp.task('test', function() {
...
+      }
+    }
+  }
+  for (var i = 0; i < pipelines.length; i++) {
+    var cb = (function(i, cb) { return function() { processStages(pipelines[i], cb); } })(i, cb);
+  }
+  return cb(null);
+}
+
+experimentTask('experiment',
```
Comment (shans) [inline gulpfile.js]: Is it worth keeping this? It's an example of how to manually set up a gulp build target using the experiment data structures.

**Comment by shans** [comment]
The basic approach is that you can set up an experiment using the syntax in "example.exp". This has two parts:
(1) a tree of steps that link target names
(2) a mapping from target names to regular expressions which will match input/output files.

The 'runExperiment' target can then be used to run the experiment, which will match against file names in the current directory and run the specified steps.

So for example, using example.exp, if I started with the following two files:
mobile-mail.json
mobile-mail-nostyle.json

then I would end up with
mobile-mail-[nuked].html
mobile-mail-[nuked, nostyle].html
mobile-mail-[nuked, compressed, nostyle].html
mobile-mail-[nuked, extracted].html

File: example.exp [hunk 2]
```diff
@@ -0,0 +1,11 @@
+input -NukeIFrameFilter-> -HTMLWriter-> case-1
+input-nostyle -NukeIFrameFilter-> -HTMLWriter-> case-2
+input-nostyle -StyleFilter-> reduced* -NukeIFrameFilter-> -HTMLWriter-> case-3
+reduced* -StyleMinimizationFilter-> -NukeIFrameFilter-> -HTMLWriter-> case-4
```
Comment (dglazkov) [inline example.exp]: Do you have a doc for this new-fangled language? :smiley:

File: gulpfile.js (hunk 1 above)
Comment (dglazkov) [inline gulpfile.js]: Put it in the doc?

**Comment by dglazkov** [comment]
LGTM. Exciting!

File: example.exp (hunk 2 above)
Comment (shans) [inline example.exp]: #25

**Pull request merged**